File: week5/PPPinheritance.cpp
#include <iostream>
using namespace std;
class Parent
{
public:
    string name = "vamsi";
protected:
int roll = 561;
private:
    int code = 33;
};
class PublicChild : public Parent
{
public:
    int getProtected()
    {
        return roll;
    }
};
class ProtectedChild : protected Parent
{
public:
    string getPublic()
    {
        return name;
    }
    int getProtected()
    {
        return roll;
    }
};
class PrivateChild : private Parent
{
public:
    string getPublic()
    {
        return name;
    }
    int getProtected()
    {
        return roll;
    }
};
int main()
{
    PublicChild obj1;
    ProtectedChild obj2;
    PrivateChild obj3;
cout << "In Public Inheritance" << endl;
    cout << "Public Member : " << obj1.name << endl;
    cout << "Protected Member : " << obj1.getProtected() << endl;
    cout << "In Protected Inheritance" << endl;
    cout << "Public Member : " << obj2.getPublic() << endl;
    cout << "Protected Member : " << obj2.getProtected() << endl;
    cout << "Private Member : " << obj2.getPrivate() << endl;
    cout << "In Private Inheritance" << endl;
    cout << "Public Member : " << obj3.getPublic() << endl;
    cout << "Protected Member : " << obj3.getProtected() << endl;
}

File: week5/PPPinheritance.java
class Public 
{ 
public 
    String publicname="vamsi"; 
private 
    int privateroll=561; 
protected 
    int protectedcode=33; 
public 
    int getPrivate()
    { 
        return privateroll; 
    } 
} 
class InheritanceAccess 
{ 
    public static void main(String[] args) 
    { 
        Public pub=new Public(); 
        System.out.print("Public : "+pub.publicname+"\n"); 
        System.out.print("Protected : "+pub.protectedcode+"\n"); 
    } 
}
